hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -106,32 +106,24 @@ returnFundsToFaucet ::
   RunningNode ->
   Actor ->
   IO ()
-returnFundsToFaucet tracer node@RunningNode{networkId, nodeSocket} sender = do
+returnFundsToFaucet tracer RunningNode{networkId, nodeSocket} sender = do
   (faucetVk, _) <- keysFor Faucet
   let faucetAddress = mkVkAddress networkId faucetVk
 
   (senderVk, senderSk) <- keysFor sender
   utxo <- queryUTxOFor networkId nodeSocket QueryTip senderVk
-
   retryOnExceptions tracer $ do
     let utxoValue = balance @Tx utxo
     let allLovelace = selectLovelace utxoValue
-    -- select tokens other than ADA here so we can burn it afterwards
-    let otherTokens = filterValue (/= AdaAssetId) utxoValue
-    -- XXX: Using a hard-coded high-enough value to satisfy the min utxo value.
-    -- NOTE: We use the faucet address as the change deliberately here.
-    fee <- calculateTxFee node senderSk utxo faucetAddress 1_000_000
-    let returnBalance = allLovelace - fee
-    tx <- sign senderSk <$> buildTxBody utxo faucetAddress returnBalance otherTokens
+    tx <- sign senderSk <$> buildTxBody utxo faucetAddress
     submitTransaction networkId nodeSocket tx
     void $ awaitTransaction networkId nodeSocket tx
-    traceWith tracer $ ReturnedFunds{actor = actorName sender, returnAmount = returnBalance}
+    traceWith tracer $ ReturnedFunds{actor = actorName sender, returnAmount = allLovelace}
  where
-  buildTxBody utxo faucetAddress lovelace otherTokens =
-    let theOutput = TxOut faucetAddress (lovelaceToValue lovelace <> negateValue otherTokens) TxOutDatumNone ReferenceScriptNone
-     in buildTransaction networkId nodeSocket faucetAddress utxo [] [theOutput] >>= \case
-          Left e -> throwIO $ FaucetFailedToBuildTx{reason = e}
-          Right body -> pure body
+  buildTxBody utxo faucetAddress =
+    buildTransaction networkId nodeSocket faucetAddress utxo [] [] >>= \case
```

Comment:
Could: add a comment why this if fine (i.e. explain that with no outputs a change output with all value is created and paid to the faucet address)